File: cpp/oop/diamondInheritance/Horse.cpp
#include <iostream>
#include <string>

#include "Horse.h"

using namespace std;

Horse::Horse(string name) :
    Animal(name)
{
    cout << "Horse::Horse: name=" << name << endl;
}

Horse::~Horse() {
    cout << "Horse::~Horse" << endl;
}

void Horse::run() {
    cout << "Horse::run: 4 legs"  << endl;
}

File: java/oop/diamondInheritance/Horse.java
public class Horse extends Animal implements ICanRunAnimal {
    public Horse(String name) {
	super(name);
	System.out.println("[D] Horse::Horse: name=[" + name + "]");
    }

    public void run() {
	System.out.println("[D] Horse::run with 4 legs");
    }
}

